src/rqt_image_view/image_view.cpp

```diff
@@ -112,12 +112,14 @@ void ImageView::initPlugin(qt_gui_cpp::PluginContext& context)
   hide_toolbar_action_->setCheckable(true);
   ui_.image_frame->addAction(hide_toolbar_action_);
   connect(hide_toolbar_action_, SIGNAL(toggled(bool)), this, SLOT(onHideToolbarChanged(bool)));
+  change_topic_service_ = getNodeHandle().advertiseService( getNodeHandle().getNamespace() + std::to_string(context.serialNumber()) + "/change_image_topic", &ImageView::changeImageService, this);
```

Comment:
I would rather use the term for the topic name as for the service name: `set_image_topic` (the same for the callback function `changeImageService`).

Also please wrap the line after `getNodeHandle().advertiseService(` for readability.